x-pack/legacy/plugins/maps/public/layers/styles/vector/components/stop_input.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+
+import { EuiComboBox } from '@elastic/eui';
+
+export class StopInput extends Component {
+  state = {
+    suggestions: [],
+    isLoadingSuggestions: false,
+    hasPrevFocus: false,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    this._loadSuggestions.cancel();
+  }
+
+  _onFocus = () => {
+    if (!this.state.hasPrevFocus) {
+      this.setState({ hasPrevFocus: true });
+      this._onSearchChange('');
+    }
+  };
+
+  _onChange = selectedOptions => {
+    this.props.onChange(_.get(selectedOptions, '[0].label', ''));
+  };
+
+  _onCreateOption = newValue => {
+    this.props.onChange(newValue);
+  };
+
+  _onSearchChange = async searchValue => {
+    this.setState(
+      {
+        isLoadingSuggestions: true,
+        searchValue,
+      },
+      () => {
+        this._loadSuggestions(searchValue);
+      }
+    );
+  };
+
+  _loadSuggestions = _.debounce(async searchValue => {
+    let suggestions = [];
+    try {
+      suggestions = await this.props.getValueSuggestions(searchValue);
+    } catch (error) {
+      // ignore suggestions error
+    }
+
+    if (this._isMounted && searchValue === this.state.searchValue) {
+      this.setState({
+        isLoadingSuggestions: false,
+        suggestions,
+      });
+    }
+  }, 300);
+
+  render() {
+    const suggestionOptions = this.state.suggestions.map(suggestion => {
+      return { label: suggestion };
+    });
+
+    const selectedOptions = [];
+    if (this.props.value) {
+      let option = suggestionOptions.find(({ label }) => {
+        return label === this.props.value;
+      });
+      if (!option) {
+        option = { label: this.props.value };
+        suggestionOptions.unshift(option);
+      }
+      selectedOptions.push(option);
+    }
+
```

Comment:
See other comments wrt IPs and Booleans.

Another approach would be to just disable auto-complete for IPs and booleans altogether iso trying to actual auto-complete those.

Reply:
Autocomplete service does not support `ip` field types. Just like the filter pill value input, I have opted to only provide suggestions for `string` and `boolean` field types.